src/main/java/uk/gov/ons/census/notifyprocessor/client/CaseClient.java

```diff
@@ -22,12 +23,17 @@ public CaseClient(RestTemplate restTemplate) {
   }
 
   public UacQidDTO getUacQid(String caseId, int questionnaireType) {
-    String url = "http://" + host + ":" + port + "/uacqid/create";
+    UriComponentsBuilder url =
+        UriComponentsBuilder.newInstance()
+            .scheme("http")
+            .host(host)
+            .port(port)
+            .path("/uacqid/create");
     CaseDetailsDTO caseDetails = new CaseDetailsDTO();
     caseDetails.setCaseId(caseId);
     caseDetails.setQuestionnaireType(Integer.toString(questionnaireType));
 
-    UacQidDTO uacQid = restTemplate.postForObject(url, caseDetails, UacQidDTO.class);
+    UacQidDTO uacQid = restTemplate.postForObject(url.toUriString(), caseDetails, UacQidDTO.class);
```

Comment:
you can just do `uriComponents.toUri()` and not pass in a string

Reply:
Code updated